drivers/net/ppp.c

```diff
@@ -875,6 +875,8 @@ static int ppp_stop(struct device *dev)
 
 	net_ppp_carrier_off(context->iface);
 
+	context->modem_init_done = false;
```

Comment:
This variable is atomic, please use the atomic api to set the value.

Reply:
addressed